features/nvstore/source/nvstore.cpp

```diff
@@ -0,0 +1,1066 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// ----------------------------------------------------------- Includes -----------------------------------------------------------
+
+#include "nvstore.h"
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+#include "mbed_critical.h"
+#include "mbed_assert.h"
+#include "Thread.h"
+#include <string.h>
+#include <stdio.h>
+
+#if NVSTORE_ENABLED
+
+// --------------------------------------------------------- Definitions ----------------------------------------------------------
+
+#define PR_ERR printf
+#define PR_INFO printf
+#define PR_DEBUG printf
+
+#define MIN(a,b)            ((a) < (b) ? (a) : (b))
+#define MAX(a,b)            ((a) > (b) ? (a) : (b))
+
+#define DELETE_ITEM_FLAG  0x8000
+#define SET_ONCE_FLAG     0x4000
+#define HEADER_FLAG_MASK  0xF000
+
+#define MASTER_RECORD_KEY 0xFFE
+#define NO_KEY            0xFFF
+
+typedef struct {
+    uint16_t key_and_flags;
+    uint16_t length;
+    uint32_t mac;
+} record_header_t __attribute__((aligned(4)));
+
+
+#define OFFS_BY_KEY_AREA_MASK     0x80000000
```

Comment:
Are these signed or unsigned , should this be `0x80000000UL` ? I can see these are used with `uint32_t` variables in this file